nautobot/core/api/utils.py

```diff
@@ -326,7 +326,7 @@ def return_nested_serializer_data_based_on_depth(serializer, depth, obj, obj_rel
     """
     Handle serialization of GenericForeignKey fields at an appropriate depth.
 
-    When depth = 0, return the URL for the related object.
+    When depth = 0, return a brief representation of the related object, containing URL, PK, and object_type.
```

Comment:
Wish we could include `composite_key` in this case too but calculating that possibly involves N additional database lookups per object, so that's to be avoided.